numpyro/distributions/constraints.py

```diff
@@ -697,6 +698,31 @@ def feasible_like(self, prototype):
         return jax.numpy.full_like(prototype, prototype.shape[-1] ** (-0.5))
 
 
+class _ZeroSum(Constraint):
+    def __init__(self, event_dim=1):
+        self.event_dim = event_dim
+        super().__init__()
+
+    def __call__(self, x):
+        jnp = np if isinstance(x, (np.ndarray, np.generic)) else jax.numpy
+        zerosum_true = []
```

Comment:
Could do something like `zerosum_true = True` and then in the loop `zerosum_true = zerosum_true & jnp.allclose(x.sum(dim), 0, ...)`. That means we don't have to keep track of the list.